mexico-city/src/mc_nitro.rs

```diff
@@ -0,0 +1,219 @@
+                let return_message: MCMessage = match cs_result {
+                    Ok(_) => MCMessage::Status(NitroStatus::Success),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetTLSDataNeeded(session_id) => {
+                println!("mc_nitro::main GetTLSDataNeeded");
+                let return_message = match managers::baja_manager::get_data_needed(session_id) {
+                    Ok(needed) => MCMessage::TLSDataNeeded(needed),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::SendTLSData(session_id, tls_data) => {
+                println!("mc_nitro::main SendTLSData");
+                let return_message = match managers::baja_manager::send_data(session_id, &tls_data) {
+                    Ok(_) => MCMessage::Status(NitroStatus::Success),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetTLSData(session_id) => {
+                println!("mc_nitro::main GetTLSData");
+                let return_message = match managers::baja_manager::get_data(session_id) {
+                    Ok((active, output_data)) => MCMessage::TLSData(output_data, active),
+                    Err(_) => MCMessage::Status(NitroStatus::Fail),
+                };
+                return_message
+            },
+            MCMessage::GetPSAAttestationToken(challenge) => {
+                println!("mc_nitro::main GetPSAAttestationToken");
+                get_psa_attestation_token(&challenge)
+                    .map_err(|err| format!("mc_nitro::nitro_main get_psa_attestation_failed:{:?}", err))?
+            },
+            MCMessage::ResetEnclave => {
+                println!("mc_nitro::main ResetEnclave");
+                MCMessage::Status(NitroStatus::Success)
+            },
+            _ => {
+                println!("mc_nitro::main Unknown Opcode");
+                MCMessage::Status(NitroStatus::Unimplemented)
+            },
+        };
+        let return_buffer = bincode::serialize(&return_message)
+            .map_err(|err| format!("mc_nitro::main serialize failed:{:?}", err))?;
+        println!("mc_nitro::main calling send buffer with buffer_len:{:?}", return_buffer.len());
+        send_buffer(fd, &return_buffer)
+            .map_err(|err| format!("mc_nitro::main send_buffer failed:{:?}", err))?;
+    }
+}
+
+fn initialize(policy_json: &String) -> Result<MCMessage, String> {
+    println!("mc_nitro::initialize started");
+    managers::baja_manager::init_baja(&policy_json)
+        .map_err(|err| format!("mc_nitro::main init_baja failed:{:?}", err))?;
+    println!("mc_nitro::main init_baja completed");
+    return Ok(MCMessage::Status(NitroStatus::Success));
```

Comment:
I think the `return` statement is superfluous and `Ok(MCMessage::Status(NitroStatus::Success))` will do.